llvm/lib/Target/AMDGPU/AMDGPUPromoteAlloca.cpp

```diff
@@ -1556,12 +1556,34 @@ bool AMDGPUPromoteAllocaImpl::tryPromoteAllocaToLDS(AllocaInst &I,
     case Intrinsic::invariant_start:
     case Intrinsic::invariant_end:
     case Intrinsic::launder_invariant_group:
-    case Intrinsic::strip_invariant_group:
+    case Intrinsic::strip_invariant_group: {
+      SmallVector<Type *> ArgTy;
+      SmallVector<Value *> Args;
+      if (Intr->getIntrinsicID() == Intrinsic::invariant_start) {
+        Value *Size = Intr->getArgOperand(0);
+        ArgTy.emplace_back(Offset->getType());
+        Args.emplace_back(Size);
+        Args.emplace_back(Offset);
+      } else if (Intr->getIntrinsicID() == Intrinsic::invariant_end) {
+        Value *InvariantPtr = Intr->getArgOperand(0);
+        Value *Size = Intr->getArgOperand(1);
+        ArgTy.emplace_back(Offset->getType());
+        Args.emplace_back(InvariantPtr);
+        Args.emplace_back(Size);
+        Args.emplace_back(Offset);
+      } else {
+        ArgTy.emplace_back(Offset->getType());
```

Comment:
This is common to all cases, you could initialize the vectors with it

Reply:
Sure, will do a follow up.